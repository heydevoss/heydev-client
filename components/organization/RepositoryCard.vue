<template>
  <div class="card">
    <el-card shadow="hover">
      <div slot="header" class="card--header">
        <span class="repository--name">{{ repository.name }}</span>
      </div>
      <div class="repository--counters">
        <i class="el-icon-star-on">{{ repository.totalStars }}</i>
        <i class="el-icon-warning">{{ repository.totalOpenIssues }}</i>
        <img width="15px" height="15px" src="/fork.png" />
        {{ repository.totalForks }}
      </div>
    </el-card>
  </div>
</template>

<script>
export default {
  name: 'Repository',
  props: {
    repository: Object
  }
}
</script>

<style lang="scss">
.card {
  min-width: 200px;
  min-height: 200px;
}
</style>
