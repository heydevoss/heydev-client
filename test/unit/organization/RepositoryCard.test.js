import { shallowMount } from '@vue/test-utils'
import RepositoryCard from '@/components/organization/RepositoryCard.vue'

describe('<RepositoryCard/>', () => {
  const wrapper = shallowMount(RepositoryCard, {
    stubs: {
      'el-card': true
    },
    propsData: {
      repository: {
        name: 'MyTestRepository',
        totalStars: 20,
        totalForks: 10,
        totalOpenIssues: 3
      }
    }
  })

  it('should be a Vue instance and renders correctly', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render repository props', () => {
    expect(wrapper.html()).toContain('MyTestRepository')
    expect(wrapper.html()).toContain('20')
    expect(wrapper.html()).toContain('10')
    expect(wrapper.html()).toContain('3')
  })

  it('It should render repository correctly when props are not passed', () => {
    wrapper.setProps({
      repository: {
        name: 'MyTestRepository'
      }
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.html()).toContain('MyTestRepository')
    wrapper.setProps({})
  })
})
