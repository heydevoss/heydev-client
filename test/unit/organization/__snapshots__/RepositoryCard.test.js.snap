// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepositoryCard/> should be a Vue instance and renders correctly 1`] = `
<div
  class="card"
>
  <el-card-stub
    shadow="hover"
  >
    <div
      class="card--header"
    >
      <span
        class="repository--name"
      >
        MyTestRepository
      </span>
    </div>
     
    <div
      class="repository--counters"
    >
      <i
        class="el-icon-star-on"
      >
        20
      </i>
       
      <i
        class="el-icon-warning"
      >
        3
      </i>
       
      <img
        height="15px"
        src="/fork.png"
        width="15px"
      />
      
      10
    
    </div>
  </el-card-stub>
</div>
`;
